types/app.d.ts

```diff
@@ -143,6 +144,13 @@ export function eventGrid(name: string, options: EventGridFunctionOptions): void
  */
 export function cosmosDB(name: string, options: CosmosDBFunctionOptions): void;
 
+/**
+ * Registers a function in your app that will be triggered when an instance is added to scale a running function app
+ * @param name The name of the function. The name must be unique within your app and will mostly be used for your own tracking purposes
```

Comment:
I think we should include more of the "considerations" listed in the warmup trigger docs. Maybe something like this:

> Registers a function in your app that will be triggered during scale-out operations.
> It's not triggered during restarts or other non-scale startups, so make sure your app can run without relying on the warmup trigger. Lazy loading is a good pattern to achieve this goal.
> The warmup trigger isn't available on the Consumption plan.

Reply:
Good point. I added more information and a link to the source as well.